Index: Backpack/src/main/java/net/skyscanner/backpack/dialog/DialogConstants.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Backpack/src/main/java/net/skyscanner/backpack/dialog/DialogConstants.kt	(revision 874807dacd2bc6be4b8a73127c2be672687faa37)
+++ Backpack/src/main/java/net/skyscanner/backpack/dialog/DialogConstants.kt	(date 1565249671000)
@@ -5,17 +5,35 @@
 
 internal class DialogConstants(val context: Context) {
 
+  // 4dp
   private val spacingSm = context.resources.getDimension(R.dimen.bpkSpacingSm)
 
+  // 72dp
   val iconContainerSize = (spacingSm * 18).toInt()
+
+  // 36
   val iconContainerRadius = (iconContainerSize / 2.0).toFloat()
+
+  // 4
   val iconContainerBorderSize = spacingSm
 
+  // 32
   val contentContainerMarginTop = iconContainerSize / 2
+
+
+  // 48
   val contentContainerPaddingTop = (contentContainerMarginTop + context.resources.getDimension(R.dimen.bpkSpacingBase)).toInt()
+
+  // 8
   val contentContainerPaddingBottom = context.resources.getDimension(R.dimen.bpkSpacingMd).toInt()
+
+  // 32dp
   val contentContainerPaddingHorizontal = context.resources.getDimension(R.dimen.bpkSpacingXl).toInt()
+
+  // 4dp
   val contentRadius = context.resources.getDimension(R.dimen.bpkBorderRadiusSm)
 
+  // 8dp
   val buttonStackItemSpace = context.resources.getDimension(R.dimen.bpkSpacingMd).toInt()
+
 }
Index: app/src/main/java/net/skyscanner/backpack/demo/stories/DialogStory.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/net/skyscanner/backpack/demo/stories/DialogStory.kt	(revision 874807dacd2bc6be4b8a73127c2be672687faa37)
+++ app/src/main/java/net/skyscanner/backpack/demo/stories/DialogStory.kt	(date 1565249671000)
@@ -39,7 +39,28 @@
       val dialog = BpkDialog(context)
       dialog.apply {
         title = "You are going to Tokyo!"
-        description = "Your flight is all booked. Why not check out some hotels now?"
+        description =
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+            ""
+
         icon = BpkDialog.Icon(
           R.drawable.bpk_tick,
           R.color.bpkGreen500
Index: Backpack/src/main/java/net/skyscanner/backpack/dialog/BpkDialog.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Backpack/src/main/java/net/skyscanner/backpack/dialog/BpkDialog.kt	(revision 874807dacd2bc6be4b8a73127c2be672687faa37)
+++ Backpack/src/main/java/net/skyscanner/backpack/dialog/BpkDialog.kt	(date 1565249671000)
@@ -87,27 +87,19 @@
 
     viewHolder = ViewHolder(view).apply {
       val params = contentLayout.layoutParams as RelativeLayout.LayoutParams
-      params.topMargin = variables.contentContainerMarginTop
-      contentLayout.background = getContentBackground()
-      contentLayout.setPadding(
-        variables.contentContainerPaddingHorizontal,
-        variables.contentContainerPaddingTop,
-        variables.contentContainerPaddingHorizontal,
-        variables.contentContainerPaddingBottom
-      )
+//      params.topMargin = variables.contentContainerMarginTop
+//      contentLayout.background = getContentBackground()
+//      contentLayout.setPadding(
+//        variables.contentContainerPaddingHorizontal,
+//        variables.contentContainerPaddingTop,
+//        variables.contentContainerPaddingHorizontal,
+//        variables.contentContainerPaddingBottom
+//      )
     }
 
     return view
   }
 
-  private fun getContentBackground(): GradientDrawable {
-    val shape = GradientDrawable()
-    shape.shape = GradientDrawable.RECTANGLE
-    val radii = variables.contentRadius
-    shape.cornerRadii = floatArrayOf(radii, radii, radii, radii, radii, radii, radii, radii)
-    shape.setColor(ResourcesCompat.getColor(context.resources, R.color.bpkWhite, context.theme))
-    return shape
-  }
 
   private class ViewHolder(container: View) {
     val title: BpkText = container.findViewById(R.id.dialog_title)
Index: Backpack/src/main/res/drawable/bpk_dialog_bg.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Backpack/src/main/res/drawable/bpk_dialog_bg.xml	(date 1565249671000)
+++ Backpack/src/main/res/drawable/bpk_dialog_bg.xml	(date 1565249671000)
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android" android:shape="rectangle">
+  <corners android:radius="@dimen/bpkBorderRadiusSm" />
+  <solid android:color="@color/bpkWhite" />
+</shape>
Index: Backpack/src/main/java/net/skyscanner/backpack/dialog/BpkDialogLayout.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Backpack/src/main/java/net/skyscanner/backpack/dialog/BpkDialogLayout.kt	(date 1565249671000)
+++ Backpack/src/main/java/net/skyscanner/backpack/dialog/BpkDialogLayout.kt	(date 1565249671000)
@@ -0,0 +1,17 @@
+package net.skyscanner.backpack.dialog
+
+import android.content.Context
+import android.util.AttributeSet
+import android.widget.FrameLayout
+
+class BpkDialogLayout @JvmOverloads constructor(
+  context: Context,
+  attrs: AttributeSet? = null,
+  defStyleAttr: Int = 0
+) : FrameLayout(context, attrs, defStyleAttr) {
+
+
+
+
+
+}
Index: Backpack/src/main/res/values/dimens.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Backpack/src/main/res/values/dimens.xml	(revision 874807dacd2bc6be4b8a73127c2be672687faa37)
+++ Backpack/src/main/res/values/dimens.xml	(date 1565249671000)
@@ -10,4 +10,5 @@
   <dimen name="bpk_horizontal_nav_size_normal">36dp</dimen>
   <dimen name="bpk_rating_score_badge_size_small">36dp</dimen>
   <dimen name="bpk_rating_score_badge_size_large">48dp</dimen>
+  <dimen name="bpk_dialog_padding_top">48dp</dimen>
 </resources>
Index: Backpack/src/main/res/layout/bpk_dialog.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Backpack/src/main/res/layout/bpk_dialog.xml	(revision 874807dacd2bc6be4b8a73127c2be672687faa37)
+++ Backpack/src/main/res/layout/bpk_dialog.xml	(date 1565249671000)
@@ -4,7 +4,7 @@
   xmlns:app="http://schemas.android.com/apk/res-auto"
   android:layout_width="match_parent"
   android:layout_height="match_parent"
-  android:background="@android:color/transparent" >
+  android:background="@drawable/bpk_dialog_bg">
 
   <androidx.appcompat.widget.LinearLayoutCompat
     android:id="@+id/dialog_content_layout"
Index: Backpack/src/main/java/net/skyscanner/backpack/toggle/BpkSwitch.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Backpack/src/main/java/net/skyscanner/backpack/toggle/BpkSwitch.kt	(revision 874807dacd2bc6be4b8a73127c2be672687faa37)
+++ Backpack/src/main/java/net/skyscanner/backpack/toggle/BpkSwitch.kt	(date 1565249671000)
@@ -2,6 +2,7 @@
 
 import android.content.Context
 import android.content.res.ColorStateList
+import android.graphics.Typeface
 import android.util.AttributeSet
 import androidx.appcompat.widget.SwitchCompat
 import net.skyscanner.backpack.R
