Index: Backpack/src/main/java/net/skyscanner/backpack/dialog/DialogConstants.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Backpack/src/main/java/net/skyscanner/backpack/dialog/DialogConstants.kt	(revision 874807dacd2bc6be4b8a73127c2be672687faa37)
+++ Backpack/src/main/java/net/skyscanner/backpack/dialog/DialogConstants.kt	(date 1565354470000)
@@ -5,17 +5,35 @@
 
 internal class DialogConstants(val context: Context) {
 
+  // 4dp
   private val spacingSm = context.resources.getDimension(R.dimen.bpkSpacingSm)
 
+  // 72dp
   val iconContainerSize = (spacingSm * 18).toInt()
+
+  // 36
   val iconContainerRadius = (iconContainerSize / 2.0).toFloat()
+
+  // 4
   val iconContainerBorderSize = spacingSm
 
+  // 32
   val contentContainerMarginTop = iconContainerSize / 2
+
+
+  // 48
   val contentContainerPaddingTop = (contentContainerMarginTop + context.resources.getDimension(R.dimen.bpkSpacingBase)).toInt()
+
+  // 8
   val contentContainerPaddingBottom = context.resources.getDimension(R.dimen.bpkSpacingMd).toInt()
+
+  // 32dp
   val contentContainerPaddingHorizontal = context.resources.getDimension(R.dimen.bpkSpacingXl).toInt()
+
+  // 4dp
   val contentRadius = context.resources.getDimension(R.dimen.bpkBorderRadiusSm)
 
+  // 8dp
   val buttonStackItemSpace = context.resources.getDimension(R.dimen.bpkSpacingMd).toInt()
+
 }
Index: app/src/main/java/net/skyscanner/backpack/demo/stories/DialogStory.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/net/skyscanner/backpack/demo/stories/DialogStory.kt	(revision 874807dacd2bc6be4b8a73127c2be672687faa37)
+++ app/src/main/java/net/skyscanner/backpack/demo/stories/DialogStory.kt	(date 1565354470000)
@@ -39,7 +39,39 @@
       val dialog = BpkDialog(context)
       dialog.apply {
         title = "You are going to Tokyo!"
-        description = "Your flight is all booked. Why not check out some hotels now?"
+        description = "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+//          "Your flight is all booked. Why not check out some hotels now? \n" +
+//          "Your flight is all booked. Why not check out some hotels now? \n" +
+//          "Your flight is all booked. Why not check out some hotels now? \n" +
+//          "Your flight is all booked. Why not check out some hotels now? \n" +
+//          "Your flight is all booked. Why not check out some hotels now? \n" +
+//          "Your flight is all booked. Why not check out some hotels now? \n" +
+//          "Your flight is all booked. Why not check out some hotels now? \n" +
+//          "Your flight is all booked. Why not check out some hotels now? \n" +
+//          "Your flight is all booked. Why not check out some hotels now? \n" +
+//          "Your flight is all booked. Why not check out some hotels now? \n" +
+//          "Your flight is all booked. Why not check out some hotels now? \n" +
+//          "Your flight is all booked. Why not check out some hotels now? \n" +
+//          "Your flight is all booked. Why not check out some hotels now? \n" +
+//          "Your flight is all booked. Why not check out some hotels now? \n" +
+//          "Your flight is all booked. Why not check out some hotels now? \n" +
+//          "Your flight is all booked. Why not check out some hotels now? \n" +
+//          "Your flight is all booked. Why not check out some hotels now? \n" +
+          ""
         icon = BpkDialog.Icon(
           R.drawable.bpk_tick,
           R.color.bpkGreen500
Index: Backpack/src/main/java/net/skyscanner/backpack/dialog/BpkDialog.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Backpack/src/main/java/net/skyscanner/backpack/dialog/BpkDialog.kt	(revision 874807dacd2bc6be4b8a73127c2be672687faa37)
+++ Backpack/src/main/java/net/skyscanner/backpack/dialog/BpkDialog.kt	(date 1565354470000)
@@ -71,7 +71,7 @@
       window?.setGravity(Gravity.BOTTOM)
       window?.setWindowAnimations(R.style.Bpk_dialog_animation)
     }
-
+    window.setLayout(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT)
     requestWindowFeature(Window.FEATURE_NO_TITLE)
     setContentView(setUpContentView())
     window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
@@ -84,20 +84,43 @@
   private fun setUpContentView(): View {
     val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
     val view = inflater.inflate(R.layout.bpk_dialog, null)
-
+  MaterialAlertDialogBuilder
     viewHolder = ViewHolder(view).apply {
-      val params = contentLayout.layoutParams as RelativeLayout.LayoutParams
-      params.topMargin = variables.contentContainerMarginTop
-      contentLayout.background = getContentBackground()
-      contentLayout.setPadding(
-        variables.contentContainerPaddingHorizontal,
-        variables.contentContainerPaddingTop,
-        variables.contentContainerPaddingHorizontal,
-        variables.contentContainerPaddingBottom
-      )
+//      val params = contentLayout.layoutParams as RelativeLayout.LayoutParams
+//      params.topMargin = variables.contentContainerMarginTop
+//      contentLayout.background = getContentBackground()
+//      contentLayout.setPadding(
+////        0,0,0,0
+//
+//        variables.contentContainerPaddingHorizontal,
+//        0,
+//        variables.contentContainerPaddingTop,
+//        variables.contentContainerPaddingHorizontal,
+//        0
+//        variables.contentContainerPaddingBottom
+//      )
     }
+
+    return view.apply {
+      viewTreeObserver.addOnPreDrawListener {
+
+
+        val measuredHeight = view.measuredHeight
+        val heightPixels = context.resources.displayMetrics.heightPixels
+        val fl = heightPixels * 0.75f
+        window.setBackgroundDrawable(ColorDrawable(
+          if (measuredHeight < fl) {
+            Color.BLUE
+          } else {
+            window.setDimAmount(0f)
+//            window.setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT)
+            Color.RED
+          }
 
-    return view
+        ))
+        true
+      }
+    }
   }
 
   private fun getContentBackground(): GradientDrawable {
Index: Backpack/src/main/res/layout/bpk_dialog.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Backpack/src/main/res/layout/bpk_dialog.xml	(revision 874807dacd2bc6be4b8a73127c2be672687faa37)
+++ Backpack/src/main/res/layout/bpk_dialog.xml	(date 1565354470000)
@@ -3,7 +3,7 @@
   xmlns:android="http://schemas.android.com/apk/res/android"
   xmlns:app="http://schemas.android.com/apk/res-auto"
   android:layout_width="match_parent"
-  android:layout_height="match_parent"
+  android:layout_height="wrap_content"
   android:background="@android:color/transparent" >
 
   <androidx.appcompat.widget.LinearLayoutCompat
@@ -12,6 +12,11 @@
     android:layout_height="wrap_content"
     android:layout_alignParentTop="true"
     android:layout_centerHorizontal="true"
+    android:layout_marginTop="32dp"
+    android:paddingTop="48dp"
+    android:paddingBottom="8dp"
+    android:paddingLeft="32dp"
+    android:paddingRight="32dp"
     android:background="@color/bpkWhite"
     android:orientation="vertical" >
 
