Index: app/src/main/java/net/skyscanner/backpack/demo/stories/DialogStory.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- app/src/main/java/net/skyscanner/backpack/demo/stories/DialogStory.kt	(revision 874807dacd2bc6be4b8a73127c2be672687faa37)
+++ app/src/main/java/net/skyscanner/backpack/demo/stories/DialogStory.kt	(date 1565360585000)
@@ -39,7 +39,35 @@
       val dialog = BpkDialog(context)
       dialog.apply {
         title = "You are going to Tokyo!"
-        description = "Your flight is all booked. Why not check out some hotels now?"
+        description = "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          "Your flight is all booked. Why not check out some hotels now? \n" +
+          ""
         icon = BpkDialog.Icon(
           R.drawable.bpk_tick,
           R.color.bpkGreen500
Index: Backpack/src/main/java/net/skyscanner/backpack/dialog/DialogContentLayout.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Backpack/src/main/java/net/skyscanner/backpack/dialog/DialogContentLayout.kt	(date 1565362642000)
+++ Backpack/src/main/java/net/skyscanner/backpack/dialog/DialogContentLayout.kt	(date 1565362642000)
@@ -0,0 +1,46 @@
+package net.skyscanner.backpack.dialog
+
+import android.content.Context
+import android.graphics.Color
+import android.graphics.drawable.ColorDrawable
+import android.util.AttributeSet
+import android.view.Gravity
+import android.widget.FrameLayout
+import net.skyscanner.backpack.util.unsafeLazy
+
+internal class DialogContentLayout @JvmOverloads constructor(
+  context: Context,
+  attrs: AttributeSet? = null,
+  defStyleAttr: Int = 0
+) : FrameLayout(context, attrs, defStyleAttr) {
+
+  private val content by unsafeLazy { getChildAt(0) }
+
+  private val contentParams: LayoutParams
+    get() = content.layoutParams.let { it as LayoutParams }
+
+  var verticalGravity: Int
+    get() = contentParams.gravity
+    set(value) {
+      val verticalGravityComponent = value and (Gravity.TOP or Gravity.BOTTOM or Gravity.CENTER_VERTICAL)
+      contentParams.gravity = verticalGravityComponent or Gravity.CENTER_HORIZONTAL
+      content.requestLayout()
+    }
+
+  override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
+    super.onMeasure(widthMeasureSpec, heightMeasureSpec)
+    if (content.measuredHeight >= measuredHeight * fullscreenDefinition) {
+      background = fullscreenBg
+    } else {
+      background = modalBg
+    }
+  }
+
+  private companion object {
+    val modalBg = ColorDrawable(Color.TRANSPARENT)
+    val fullscreenBg = ColorDrawable(Color.WHITE)
+    const val fullscreenDefinition = 0.75f
+  }
+
+
+}
Index: Backpack/src/main/res/layout/bpk_dialog.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Backpack/src/main/res/layout/bpk_dialog.xml	(revision 874807dacd2bc6be4b8a73127c2be672687faa37)
+++ Backpack/src/main/res/layout/bpk_dialog.xml	(date 1565362804000)
@@ -1,42 +1,47 @@
 <?xml version="1.0" encoding="utf-8"?>
-<RelativeLayout
+<net.skyscanner.backpack.dialog.DialogContentLayout
   xmlns:android="http://schemas.android.com/apk/res/android"
   xmlns:app="http://schemas.android.com/apk/res-auto"
   android:layout_width="match_parent"
-  android:layout_height="match_parent"
-  android:background="@android:color/transparent" >
+  android:layout_height="match_parent">
+
+  <RelativeLayout
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content">
 
-  <androidx.appcompat.widget.LinearLayoutCompat
-    android:id="@+id/dialog_content_layout"
-    android:layout_width="match_parent"
-    android:layout_height="wrap_content"
-    android:layout_alignParentTop="true"
-    android:layout_centerHorizontal="true"
-    android:background="@color/bpkWhite"
-    android:orientation="vertical" >
+    <androidx.appcompat.widget.LinearLayoutCompat
+      android:id="@+id/dialog_content_layout"
+      android:layout_width="match_parent"
+      android:layout_height="wrap_content"
+      android:layout_alignParentTop="true"
+      android:layout_centerHorizontal="true"
+      android:background="@color/bpkWhite"
+      android:orientation="vertical">
 
-    <net.skyscanner.backpack.text.BpkText
-      android:id="@+id/dialog_title"
-      android:layout_width="match_parent"
-      android:layout_height="wrap_content"
-      android:textAlignment="center"
-      app:textStyle="xl"
-      app:weight="emphasized" />
+      <net.skyscanner.backpack.text.BpkText
+        android:id="@+id/dialog_title"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:textAlignment="center"
+        app:textStyle="xl"
+        app:weight="emphasized" />
 
-    <net.skyscanner.backpack.text.BpkText
-      android:id="@+id/dialog_description"
-      android:layout_marginTop="@dimen/bpkSpacingMd"
-      android:layout_marginBottom="@dimen/bpkSpacingLg"
-      android:layout_width="match_parent"
-      android:layout_height="wrap_content"
-      android:textAlignment="center"
-      app:textStyle="lg" />
-  </androidx.appcompat.widget.LinearLayoutCompat>
+      <net.skyscanner.backpack.text.BpkText
+        android:id="@+id/dialog_description"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="@dimen/bpkSpacingMd"
+        android:layout_marginBottom="@dimen/bpkSpacingLg"
+        android:textAlignment="center"
+        app:textStyle="lg" />
+    </androidx.appcompat.widget.LinearLayoutCompat>
 
-  <net.skyscanner.backpack.dialog.BpkDialogIcon
-    android:id="@+id/dialog_icon"
-    android:layout_width="wrap_content"
-    android:layout_height="wrap_content"
-    android:layout_alignParentTop="true"
-    android:layout_centerHorizontal="true" />
-</RelativeLayout>
+    <net.skyscanner.backpack.dialog.BpkDialogIcon
+      android:id="@+id/dialog_icon"
+      android:layout_width="wrap_content"
+      android:layout_height="wrap_content"
+      android:layout_alignParentTop="true"
+      android:layout_centerHorizontal="true" />
+  </RelativeLayout>
+
+</net.skyscanner.backpack.dialog.DialogContentLayout>
Index: Backpack/src/main/java/net/skyscanner/backpack/dialog/BpkDialog.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Backpack/src/main/java/net/skyscanner/backpack/dialog/BpkDialog.kt	(revision 874807dacd2bc6be4b8a73127c2be672687faa37)
+++ Backpack/src/main/java/net/skyscanner/backpack/dialog/BpkDialog.kt	(date 1565362497000)
@@ -67,14 +67,18 @@
   }
 
   private fun setupDialog() {
-    if (style == Style.BOTTOM_SHEET) {
-      window?.setGravity(Gravity.BOTTOM)
-      window?.setWindowAnimations(R.style.Bpk_dialog_animation)
-    }
+
 
     requestWindowFeature(Window.FEATURE_NO_TITLE)
     setContentView(setUpContentView())
+    window?.setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT)
     window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
+    if (style == Style.BOTTOM_SHEET) {
+      viewHolder?.container?.verticalGravity = Gravity.BOTTOM
+      window?.setWindowAnimations(R.style.Bpk_dialog_animation)
+    } else {
+      viewHolder?.container?.verticalGravity = Gravity.CENTER_VERTICAL
+    }
 
     val metrics = DisplayMetrics()
     val windowManager = context.getSystemService(Context.WINDOW_SERVICE) as WindowManager
@@ -85,7 +89,7 @@
     val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
     val view = inflater.inflate(R.layout.bpk_dialog, null)
 
-    viewHolder = ViewHolder(view).apply {
+    viewHolder = ViewHolder(view as DialogContentLayout).apply {
       val params = contentLayout.layoutParams as RelativeLayout.LayoutParams
       params.topMargin = variables.contentContainerMarginTop
       contentLayout.background = getContentBackground()
@@ -109,7 +113,7 @@
     return shape
   }
 
-  private class ViewHolder(container: View) {
+  private class ViewHolder(val container: DialogContentLayout) {
     val title: BpkText = container.findViewById(R.id.dialog_title)
     val description: BpkText = container.findViewById(R.id.dialog_description)
     val iconView: BpkDialogIcon = container.findViewById(R.id.dialog_icon)
